rust_src/src/util.rs

```diff
@@ -13,3 +15,40 @@ pub extern "C" fn clip_to_bounds(lower: isize, num: EmacsInt, upper: isize) -> i
         upper
     }
 }
+
+/// Get boundaries of the current buffer.
+/// If BYTE is true, return byte positions.
+/// If RESTRICTED is true, limit the bounds by the buffer's visible part.
+pub fn current_buffer_bounds(byte: bool, restricted: bool) -> (isize, isize) {
```

Comment:
I agree with Sean; this isn't great. Instead of boolean arguments, just split it into four functions. That makes them easy to understand, and lets them have names that mean something. Also, instead of "restricted", you could be more specific and call it "visible".

Reply:
I was thinking the same thing. Named functions would be more clear